File: back-end/src/main/java/com/huyphan/dtos/ItemDto.java
package com.huyphan.dtos;

import lombok.Getter;
import lombok.Setter;

/** Item DTO. */
@Getter
@Setter
public abstract class ItemDto {
	/** Book id of this item. */
	private int bookId;

	/** The amount of this item. */
	private int amount;

	/** The book of this item. */
	private BookDto book;
}

File: front-end/src/app/core/services/dtos/item-dto.ts
import { BookDto } from './book-dto';

/** Item DTO. */
export interface ItemDto {
  /** Book id of this item. */
  readonly bookId: number;

  /** The amount of this item. */
  readonly amount: number;

  /** The book of this item. */
  readonly book: BookDto;
}
